tests/UnitTests/Notifications/Managers/SubscriptionManagerTests.cs

```diff
@@ -0,0 +1,1056 @@
+
+			Assert.That(subscription, Is.Not.Null);
+			Assert.That(subscription!.AlreadySeenIds, Has.Exactly(5).Items);
+			Assert.That(subscription!.AlreadySeenIds, Is.EqualTo(expected));
+		}
+
+		[Test]
+		public async Task SubscriptionManager_UpdateChannelName_UpdatesCorrectly_WhenEntryNotNull()
+		{
+			// arrange
+
+			var mockRepo = new MockNotificationRepository();
+			var mockRepoFactory = new MockNotificationRepositoryFactory(mockRepo);
+
+			const string firstChannelName = "Captain Smeghead";
+			const string secondChannelName = "Captain Smeghead 2";
+
+			const string channelId = "1";
+
+			var manager = new SubscriptionManager(null!, new NullLogger<SubscriptionManager>(), mockRepoFactory, null!, null!);
+
+			await mockRepo.AddSubscriptionAsync(channelId, DateTime.Now, "1", firstChannelName);
+			
+			// act
+
+			await manager.UpdateChannelNameAsync(secondChannelName, channelId);
+			var subscription = await mockRepo.GetSubscriptionByIdOrDefaultAsync(channelId);
+
+			// assert
+
+			Assert.That(subscription, Is.Not.Null);
+			Assert.That(subscription!.ChannelTitle, Is.EqualTo(secondChannelName));
+		}
+		
+		
+		[Test]
+		public async Task SubscriptionManager_UpdateChannelName_UpdatesCorrectly_WhenEntryHasSame()
+		{
+			// arrange
+
+			var mockRepo = new MockNotificationRepository();
+			var mockRepoFactory = new MockNotificationRepositoryFactory(mockRepo);
+
+			const string firstChannelName = "Captain Smeghead";
+			const string secondChannelName = "Captain Smeghead";
+
+			const string channelId = "1";
+
+			var manager = new SubscriptionManager(null!, new NullLogger<SubscriptionManager>(), mockRepoFactory, null!, null!);
+
+			await mockRepo.AddSubscriptionAsync(channelId, DateTime.Now, "1", firstChannelName);
+			
+			// act
+
+			await manager.UpdateChannelNameAsync(secondChannelName, channelId);
+			var subscription = await mockRepo.GetSubscriptionByIdOrDefaultAsync(channelId);
+
+			// assert
+
+			Assert.That(subscription, Is.Not.Null);
+			Assert.That(subscription!.ChannelTitle, Is.EqualTo(secondChannelName));
+		}
+		
+		[Test]
+		public async Task SubscriptionManager_UpdateChannelName_UpdatesCorrectly()
+		{
+			// arrange
+
+			var mockRepo = new MockNotificationRepository();
+			var mockRepoFactory = new MockNotificationRepositoryFactory(mockRepo);
+			
+			const string secondChannelName = "Captain Smeghead 2";
+
+			const string channelId = "1";
+
+			var manager = new SubscriptionManager(null!, new NullLogger<SubscriptionManager>(), mockRepoFactory, null!, null!);
+
+			// act
+
+			await manager.UpdateChannelNameAsync(secondChannelName, channelId);
+			var subscription = await mockRepo.GetSubscriptionByIdOrDefaultAsync(channelId);
+
+			// assert
+
+			Assert.That(subscription, Is.Null);
+		}
+
+		internal class YoutubeSubscriptionComparer : IEqualityComparer<YoutubeSubscription>
+		{
+
+			public bool Equals(YoutubeSubscription x, YoutubeSubscription y)
```

Comment:
As the CodeQL warning says, those should be nullable.
```suggestion
			public bool Equals(YoutubeSubscription? x, YoutubeSubscription? y)
```

Reply:
Also update the code below to account for nullability.